controllers/scheduler.go

```diff
@@ -441,11 +441,23 @@ func (s *Scheduler) getDestinationsForWorkloadGroup(destinationSelectors map[str
 		lo.LabelSelector = selector
 	}
 
-	err := s.Client.List(context.Background(), destinations, lo)
+	err := s.Client.List(context.Background(), destinationList, lo)
 	if err != nil {
 		s.Log.Error(err, "Error listing available Destinations")
 	}
-	return destinations.Items
+
+	if len(destinationSelectors) > 0 {
+		return destinationList.Items
+	}
+
+	destinations := []platformv1alpha1.Destination{}
```

Comment:
Just to confirm my understanding:
- Should a user set `StrictMatchLabels: true` on a destination, this destination will not be returned as a potential destination by `getDestinationsForWorkloadGroup`
- Should the user want to schedule work to this destination, it would have been specified via a `DestinationSelector` and `getDestinationsForWorkloadGroup` would have returned before even checking whether the `StrictMatchLabels: true` existed on any destinations

Reply:
hm... not sure i understood your understanding 😆

> Should a user set `StrictMatchLabels: true `on a destination, this destination will not be returned as a potential destination by `getDestinationsForWorkloadGroup`

It could be returned if the labels match 🤔

> Should the user want to schedule work to this destination, it would have been specified via a `DestinationSelector` and `getDestinationsForWorkloadGroup` would have returned before even checking whether the `StrictMatchLabels: true` existed on any destinations

yeah
basically if there are `destinationSelectors` and `Destination A` was selected, it means the labels on A match the `destinationSelectors` so no need to check if "strict" is set to true.

Reply:
Exactly this, I inverted the logic as I started with looking and this snippet and then validated how that destination _could_ have been chosen if desired (which is part of the logic earlier in the function).